Завершаем работу над проектом

M2rk13 opened the pull request:
Множественные улучшения по функционалу и удобству чтения.

---
:mortar_board: [Завершаем работу над проектом](https://up.htmlacademy.ru/php/12/user/1415187/tasks/17)

M2rk13 committed:
Переработан весь проект

Множественные улучшения по функционалу и удобству чтения.

M2rk13 committed:
-

M2rk13 committed:
Исправлена опечатка в схеме БД

00dev commented on templates/posts/post-details.php:
```diff
@@ -0,0 +1,131 @@
+            <a class="user__button user__button--subscription button button--main" href="subscribe.php?id=<?= $author['id'] ?>"><?= $user['subscribed'] ? 'Отписаться' : 'Подписаться' ?></a>
+            <a class="user__button user__button--writing button button--green" href="message.php">Сообщение</a>
+          </div>
+          <? endif; ?>
```
Короткий тег

00dev commented on reg.php:
```diff
@@ -1,42 +1,69 @@
+    'errors' => [],
+];
+
+$img_folder = __DIR__ . '\\img\\';
```
Работоспособность этого значения зависит от операционной системы. Если необходимо поддерживать разные системы, стОит использовать константу DIRECTORY_SEPARATOR в качестве разделителя (вместо \\)

00dev commented on templates/posts/link-details.php:
```diff
@@ -0,0 +1,15 @@
+<div class="popular">
+  <div class="post-link__wrapper">
+    <a class="post-link__external" href="http://<?= htmlspecialchars($post['content']) ?>" title="Перейти по ссылке">
```
Не очень хорошо подставлять схему ссылки. Тем более фиксированно. Представьте что сайт работает только через https. Тогда если в форму вставим https://yandex.ru получим http://https://yandex.ru

00dev commented on templates/posts/link-post.php:
```diff
@@ -0,0 +1,13 @@
+<div class="post-link__wrapper">
+  <a class="post-link__external" href="http://<?= htmlspecialchars($post['content']) ?>" title="Перейти по ссылке">
```
См. link-details.php

00dev commented on templates/posts/video-details.php:
```diff
@@ -0,0 +1,12 @@
+    <?= $post['youtube_url'] ? embed_youtube_video($post['youtube_url']) : '' ?>
+    <img src="img/coast-medium.jpg" alt="Превью к видео" width="360" height="188">
+  </div>
+  <a href="post-details.php?id=?<?= $post['id'] ?>" class="post-video__play-big button">
```
После id= лишний вопрос и post-details.php в корневой директории не существует

00dev commented on functions.php:
```diff
@@ -598,8 +445,544 @@ function db_connect(string $host,string $user,string $pass,string $db) {
+
+    if ($post_type == 'video') {
+        array_push($fields, 'youtube_url');
+        array_push($parameters, $post['youtube_url']);
```
В форме параметр называется video-url

00dev commented on templates/posts/text-post.php:
```diff
@@ -0,0 +1,4 @@
+<p><?=htmlspecialchars(cut_text($post['content']))?></p>
+<?php if (mb_strlen(htmlspecialchars($post['content'])) > 300) : ?>
+<a class="post-text__more-link" href="post.php?id=?<?= $post['id'] ?>">Читать далее</a>
```
Лишний вопрос после id=

00dev commented on templates/posts/video-post.php:
```diff
@@ -0,0 +1,12 @@
+<div class="post-video__block">
+   <div class="post-video__preview">
+      <?= embed_youtube_cover(htmlspecialchars($post['content'])) ?>
```
Ссылка находится в другом поле таблицы - youtube_url

00dev commented on functions.php:
```diff
@@ -598,8 +445,544 @@ function db_connect(string $host,string $user,string $pass,string $db) {
+        INNER JOIN users ON posts.author_id=users.id
+        INNER JOIN content_types ON posts.post_type=content_types.id
+        INNER JOIN subscribe ON posts.author_id = subscribe.author_id
+        LEFT JOIN (SELECT post_id, COUNT(*) AS like_count
```
Тут не нужен JOIN. Вместо `COALESCE(like_count, 0) AS likes,` можно написать сразу подзапрос `(SELECT COUNT(*) FROM likes WHERE likes.post_id =posts.post_id)`
То же самое про количество комментариев

M2rk13 committed:
Косметические улучшения + незначительные функциональные

Код подогнан под формат PSR-2
Улучшена валидация ссылки
Исправлены недочеты (короткие теги, лишние плейсхолдеры, улучшены запросы)

M2rk13 committed:
-

M2rk13 committed:
-
